src/meteors/utils/utils.py

```diff
@@ -43,3 +43,106 @@ def change_dtype_of_list(original_list: list[Any] | Any, dtype: Callable[[Any],
         return [dtype(element) for element in original_list]
     else:
         return [dtype(original_list)]
+
+
+def adjust_shape(target: torch.Tensor, source: torch.Tensor) -> torch.Tensor:
+    """Adjust the shape of a source tensor to match the shape of target tensor.
+
+    Args:
+        target (torch.Tensor): The target tensor.
+        source (torch.Tensor): The source tensor.
+
+    Returns:
+        torch.Tensor: The source tensor with the shape adjusted to match the target tensor.
+    """
+    if source.shape == target.shape:
+        return source
+
+    # If the source tensor has fewer dimensions than the target tensor, add dimensions to the source tensor
+    while source.dim() < target.dim():
+        source = source.unsqueeze(0)
+
+    # If the source tensor has more dimensions than the target tensor, remove dimensions from the source tensor
+    while source.dim() > target.dim():
+        source = source.squeeze(0)
+
+    # If the source tensor has a different shape than the target tensor, broadcast the source tensor to match the target tensor
+    if source.shape != target.shape:
+        source = source.expand_as(target)
+
+    return source
+
+
+def agg_segmentation_postprocessing(
+    soft_labels: bool = False,
+    classes_numb: int = 0,
+    class_axis: int = 1,
+    use_mask: bool = True,
+) -> Callable[[torch.Tensor, torch.Tensor], torch.Tensor]:  # pragma: no cover
+    """Postprocessing function for aggregating segmentation outputs.
+
+    This function takes the segmentation model output and sums the pixel scores for
```

Comment:
this is the description of an inner function - not the generating one. I'd move (or copy) this description into the body of the inner function and write something about the generator as well